app/src/main/java/com/example/todolistproject/repository/TaskRepository.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.todolistproject.repository
+
+import androidx.lifecycle.LiveData
+import com.example.todolistproject.database.dao.TaskDao
+import com.example.todolistproject.model.Task
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TaskRepository @Inject constructor(private val mTaskDao: TaskDao) {
+
+    fun getTaskList(): LiveData<List<Task>> {
+        return mTaskDao.getTaskList()
+    }
+
+    fun getIncompleteTaskList(): LiveData<List<Task>> {
+        return mTaskDao.getIncompleteTaskList()
+    }
+
+    fun getCompletedTaskList(): LiveData<List<Task>> {
+        return mTaskDao.getCompletedTaskList()
+    }
+
+    suspend fun insertTask(task: Task) {
+        withContext(Dispatchers.IO) {
+            mTaskDao.insertTask(task)
+        }
```

Comment:
```suggestion
            mTaskDao.insertTask(task)
```
coroutines을 이용한 어노태이션 프로세서 구성에서는 생성되는 코드에 IO Thread가 구성되어 있습니다.
아래 두개도 마찬가지입니다.

Reply:
Dao에 suspend function 을 선언하면 자동으로 background 쓰레드에서 작동한다는 말씀이신가요? 아래 블로그에서 찾았는데 말씀하신 부분이 이거에 해당하는지 헷갈리네요.

"Room 2.1 adds support for Kotlin coroutines. DAO methods can now be marked as suspending to ensure that they are not executed on the main thread. Read on to see how to use this, how it works under the hood and how to test this new functionality."

https://medium.com/androiddevelopers/room-coroutines-422b786dc4c5